Applications/Spire/Source/Ui/DecimalSpinBox.cpp

```diff
@@ -0,0 +1,46 @@
+#include "Spire/Ui/DecimalSpinBox.hpp"
+
+using namespace boost::signals2;
+using namespace Spire;
+
+DecimalSpinBox::DecimalSpinBox(double value, QWidget* parent)
+    : QAbstractSpinBox(parent) {
+  m_spin_box = new RealSpinBox(static_cast<long double>(value),
```

Comment:
What is this cast for?

Reply:
In this instance the construction of the Real is ambiguous because it can be either a long double, boost::ulong_long_type, or boost::long_long_type.

Reply:
What error message do you get when you remove the cast?

Reply:
`1>C:\Development\nexus\Applications\Spire\Dependencies\boost_1_72_0\include\boost-1_72\boost/multiprecision/cpp_dec_float.hpp(257,1): error C2593: 'operator =' is ambiguous (compiling source file ..\..\Source\Ui\DecimalSpinBox.cpp)
1>C:\Development\nexus\Applications\Spire\Dependencies\boost_1_72_0\include\boost-1_72\boost/multiprecision/cpp_dec_float.hpp(436,19): message : could be 'boost::multiprecision::backends::cpp_dec_float<15,int32_t,void> &boost::multiprecision::backends::cpp_dec_float<15,int32_t,void>::operator =(long double)' (compiling source file ..\..\Source\Ui\DecimalSpinBox.cpp)
1>C:\Development\nexus\Applications\Spire\Dependencies\boost_1_72_0\include\boost-1_72\boost/multiprecision/cpp_dec_float.hpp(430,19): message : or       'boost::multiprecision::backends::cpp_dec_float<15,int32_t,void> &boost::multiprecision::backends::cpp_dec_float<15,int32_t,void>::operator =(boost::ulong_long_type)' (compiling source file ..\..\Source\Ui\DecimalSpinBox.cpp)
1>C:\Development\nexus\Applications\Spire\Dependencies\boost_1_72_0\include\boost-1_72\boost/multiprecision/cpp_dec_float.hpp(418,19): message : or       'boost::multiprecision::backends::cpp_dec_float<15,int32_t,void> &boost::multiprecision::backends::cpp_dec_float<15,int32_t,void>::operator =(boost::long_long_type)' (compiling source file ..\..\Source\Ui\DecimalSpinBox.cpp)
1>C:\Development\nexus\Applications\Spire\Dependencies\boost_1_72_0\include\boost-1_72\boost/multiprecision/cpp_dec_float.hpp(257,1): message : while trying to match the argument list '(boost::multiprecision::backends::cpp_dec_float<15,int32_t,void>, const F)'
1>        with`

Reply:
What happens if you try ```static_cast<Real>(value)```.

Reply:
Same error.